### PR TITLE
Update transport docs and settings for changes

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This is related to #36652. In 7.0 we plan to deprecate a number of
settings that make reference to the concept of a `tcp transport`. We
mostly just have a single transport type now (based on tcp). Settings
should only reference tcp if they are referring to socket options. This
commit updates the settings in the docs. And removes string usages of
the old settings. Additionally it adds a missing remote compress setting
to the docs.

### CODE REVIEW

**Review by original-brownbear — APPROVED** [review]
LGTM

File: docs/reference/modules/remote-clusters.asciidoc [hunk 1]
```diff
@@ -161,6 +161,14 @@ PUT _cluster/settings
   are sent according to the global `transport.ping_schedule` setting, which
   defaults to ``-1` meaning that pings are not sent.
 
+`cluster.remote.${cluster_alias}.transport.compress`::
+
+  Per cluster boolean setting that allows compression to be configured for
```
Comment (lcawl) [inline docs/reference/modules/remote-clusters.asciidoc]:
```suggestion
  Per cluster boolean setting that enables you to configure compression for
```

File: docs/reference/modules/remote-clusters.asciidoc [hunk 2]
```diff
@@ -161,6 +161,14 @@ PUT _cluster/settings
...
+  requests to a specific remote cluster. This setting only impacts requests
```
Comment (lcawl) [inline docs/reference/modules/remote-clusters.asciidoc]:
```suggestion
  requests to a specific remote cluster. This setting impacts only requests
```

File: docs/reference/modules/remote-clusters.asciidoc [hunk 3]
```diff
@@ -161,6 +161,14 @@ PUT _cluster/settings
...
+  sent to the remote cluster. Elasticsearch will compress a response if the
```
Comment (lcawl) [inline docs/reference/modules/remote-clusters.asciidoc]:
```suggestion
  sent to the remote cluster. If the
```

File: docs/reference/modules/remote-clusters.asciidoc [hunk 4]
```diff
@@ -161,6 +161,14 @@ PUT _cluster/settings
...
+  inbound request was compressed. If unset, the global `transport.compress`
```
Comment (lcawl) [inline docs/reference/modules/remote-clusters.asciidoc]:
```suggestion
  inbound request is compressed, Elasticsearch compresses the response. If unset, the global `transport.compress`
```

File: docs/reference/modules/transport.asciidoc [hunk 5]
```diff
@@ -15,21 +15,21 @@ being the ideal solution for scatter (broadcast) / gather operations such
 as search in Elasticsearch.
 
 [float]
-=== TCP Transport
+=== Transport Settings
 
-The TCP transport is an implementation of the transport module using
-TCP. It allows for the following settings:
+The internal transport communicates over TCP. It allows for the following
```
Comment (lcawl) [inline docs/reference/modules/transport.asciidoc]:
```suggestion
The internal transport communicates over TCP. You can configure it with the following
```

**Comment by tbrooks8** [comment]
I marked this for 6.7 too. The new settings exist starting 6.6. So it seems reasonable to update the doc for the new settings in 6.7.

**Review by lcawl — APPROVED** [review]
Docs LGTM.  Thanks for making those changes!

**Pull request merged**